### PR TITLE
OE0-26: Patient: Replace link with contact for insuree relationship

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OE0-26

PR includes changes from feature/OE0-21 from that PR regarding address. https://openimis.atlassian.net/browse/OE0-21

### CODE REVIEW

File: api_fhir_r4/mapping/patientMapping.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+class ReliatonshipMapping(object):
```
Comment (dragos-dobre) [inline api_fhir_r4/mapping/patientMapping.py:2]: Fix spelling.

File: api_fhir_r4/mapping/patientMapping.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    reliatonship = {
```
Comment (dragos-dobre) [inline api_fhir_r4/mapping/patientMapping.py:4]: Fix spelling.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: api_fhir_r4/mapping/patientMapping.py (hunk 1 above)
Comment (sniedzielski) [reply api_fhir_r4/mapping/patientMapping.py:2]: ok

File: api_fhir_r4/mapping/patientMapping.py (hunk 2 above)
Comment (sniedzielski) [reply api_fhir_r4/mapping/patientMapping.py:4]: ok

**Comment by sniedzielski** [comment]
@dragos-dobre You can re-check after fixing spelling. thx in advance!

File: api_fhir_r4/converters/patientConverter.py [hunk 3]
```diff
@@ -328,12 +329,19 @@ def build_fhir_addresses(cls, fhir_patient, imis_insuree):
 
                 # municipality extension
                 extension = Extension.construct()
-                extension.url = "https://openimis.github.io/openimis_fhir_r4_ig/StructureDefinition/municipality"
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition-address-municipality"
```
Comment (dragos-dobre) [inline api_fhir_r4/converters/patientConverter.py:332]: Please look on last specification. The URL is StructureDefinition/address-municipality. This is generic URL for FHIR. Please change to comply with IG. The same for other URLs.

File: api_fhir_r4/converters/patientConverter.py [hunk 4]
```diff
@@ -328,12 +329,19 @@ def build_fhir_addresses(cls, fhir_patient, imis_insuree):
...
                 extension.valueString = imis_family.location.parent.name
                 family_address.extension = [extension]
 
+                # address location reference extension
+                extension = Extension.construct()
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition-address-location-reference"
```
Comment (dragos-dobre) [inline api_fhir_r4/converters/patientConverter.py:338]: Please update URL.

File: api_fhir_r4/converters/patientConverter.py [hunk 5]
```diff
@@ -350,12 +358,19 @@ def build_fhir_addresses(cls, fhir_patient, imis_insuree):
 
                 # municipality extension
                 extension = Extension.construct()
-                extension.url = "https://openimis.github.io/openimis_fhir_r4_ig/StructureDefinition/municipality"
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition-address-municipality"
```
Comment (dragos-dobre) [inline api_fhir_r4/converters/patientConverter.py:361]: Please update URL.

File: api_fhir_r4/converters/patientConverter.py [hunk 6]
```diff
@@ -350,12 +358,19 @@ def build_fhir_addresses(cls, fhir_patient, imis_insuree):
...
                 extension.valueString = imis_insuree.current_village.parent.name
                 current_address.extension = [extension]
 
+                # address location reference extension
+                extension = Extension.construct()
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition-address-location-reference"
```
Comment (dragos-dobre) [inline api_fhir_r4/converters/patientConverter.py:367]: Please update URL.

File: api_fhir_r4/converters/patientConverter.py [hunk 7]
```diff
@@ -464,15 +479,32 @@ def build_poverty_status_extension(cls, imis_insuree):
...
-        if imis_insuree.relationship and imis_insuree.family and imis_insuree.family.head_insuree:
-            fhir_link.type = imis_insuree.relationship.relation
-            fhir_link.other = PatientConverter.build_fhir_resource_reference(imis_insuree.family.head_insuree, reference_type=reference_type)
-            if type(fhir_patient.link) is not list:
-                fhir_patient.link = [fhir_link]
+    def build_fhir_contact(cls, fhir_patient, imis_insuree):
+        fhir_contact = PatientContact.construct()
+        if imis_insuree.relationship is not None and imis_insuree.family is not None \
+                and imis_insuree.family.head_insuree is not None:
+            system = f"{GeneralConfiguration.get_system_base_url()}ValueSet-patient-contact-relationship"
```
Comment (dragos-dobre) [inline api_fhir_r4/converters/patientConverter.py:486]: Please update URL.

File: api_fhir_r4/mapping/patientMapping.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+
+class RelationshipMapping(object):
+
+    relationship = {
+        "Spouse": "Spouse",
```
[the remark this answers is not in the document]
Comment (dragos-dobre) [reply api_fhir_r4/mapping/patientMapping.py:5]: Please fix this.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: api_fhir_r4/mapping/patientMapping.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (sniedzielski) [reply api_fhir_r4/mapping/patientMapping.py:5]: Ok. i will fix this.

File: api_fhir_r4/converters/patientConverter.py (hunk 4 above)
Comment (sniedzielski) [reply api_fhir_r4/converters/patientConverter.py:338]: ok

File: api_fhir_r4/converters/patientConverter.py (hunk 5 above)
Comment (sniedzielski) [reply api_fhir_r4/converters/patientConverter.py:361]: ok

File: api_fhir_r4/converters/patientConverter.py (hunk 6 above)
Comment (sniedzielski) [reply api_fhir_r4/converters/patientConverter.py:367]: ok

File: api_fhir_r4/converters/patientConverter.py (hunk 7 above)
Comment (sniedzielski) [reply api_fhir_r4/converters/patientConverter.py:486]: ok

File: api_fhir_r4/converters/patientConverter.py (hunk 3 above)
Comment (sniedzielski) [reply api_fhir_r4/converters/patientConverter.py:332]: ok. I will update this and the rest of urls.

**Review by dragos-dobre — APPROVED** [review]

**Pull request merged**